core/src/test/java/nl/nn/adapterframework/processors/InputOutputPipeProcessorTest.java

```diff
@@ -155,4 +160,205 @@ public void testRestoreMovedElementReader() throws Exception {
 	public void testRestoreMovedElementByteArray() throws Exception {
 		testRestoreMovedElement("ReplacedValue".getBytes());
 	}
+
+	@Test
+	public void testSkipWhenInputSessionKeyNotSet() throws Exception {
+		// Arrange
+		FixedResultPipe pipe = new FixedResultPipe();
+		pipe.setOnlyIfSessionKey("this-key-isnt-there");
+		pipe.setGetInputFromSessionKey("this-key-isnt-there");
+		pipe.setReturnString("not this");
+		PipeForward forward = new PipeForward();
+		forward.setName("success");
```

Comment:
You can move these 3 or 4 lines to the `@BeforeEach`, since they are almost always needed:
```
FixedResultPipe pipe = new FixedResultPipe(); //rename pipe to not clash
PipeForward forward = new PipeForward();
forward.setName("success");
pipe.registerForward(forward);
```

Reply:
You're right, I didn't yet look to optimise that but copy-pasted a single test into many!